lib/writer.h

```diff
@@ -0,0 +1,21 @@
+#ifndef WRITER_H
+#define WRITER_H
+
+#define MAX_VALUE_LEN 512
+
+#define EXEC_MODE_RECORD 0
+#define EXEC_MODE_REPLAY 1
+
+struct logline {
+    int id;
+    int event_type;
+    char value[MAX_VALUE_LEN];
```

Comment:
will prob add a timestamp field here, as well as a "writer_id" type of thing that could be a PID or something of that nature.